libraries/Microsoft.Bot.Connector/Authentication/DelegatingCredentialProvider.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Bot.Connector.Authentication
+{
+    /// <summary>
+    /// This is just an internal class to allow pre-existing implementation of the request validation to be used with a IServiceClientCredentialFactory.
+    /// </summary>
+    internal class DelegatingCredentialProvider : ICredentialProvider
+    {
+        private ServiceClientCredentialsFactory _credentialFactory;
```

Comment:
Incidentally this is a temporary internal shim piece of code to allow temporary reuse of the existing code.

ICredentialProvider should eventually go away.

But having said that, this might ship for a release. I can fix this stuff, its not a problem.